kernel/arch/dreamcast/hardware/asic.c

```diff
@@ -110,26 +113,46 @@
 #define ASIC_EVT_REGS 3
 #define ASIC_EVT_REG_HNDS 32
 
+typedef struct {
+    asic_evt_handler hdl;
+    void *data;
+} asic_evt_handler_entry_t;
+
+struct asic_thdata {
```

Comment:
Is there a compelling reason to do it this way? In this way it's not possible to determine whether or not a given handler is threaded or not (and if we need to call its own free/remove) or not. By expanding the `asic_evt_handler_entry_t` to have some component (maybe flags or a pointer for asic_thdata) the management of the handlers could be more easily be managed. 

I presume some sort of change to this with my comment at the end. But am open to having missed why this would be better or that's really unnecessary.

Reply:
My reason was keeping RAM usage low, but it's probably a better idea to add a field, yes.

Reply:
Checking against the dispatch function being installed makes fine sense instead of an additional field. If it ever becomes an issue to need to know elsewhere, a function could be exposed to make that same check.